### PR TITLE
Remove NMO from HCO and set it as a dependency to HCO at OLM level

### PR DESCRIPTION
**Pull request by orenc1**
When subscribing to kubevirt-hyperconverged operator, OLM will also deploy the latest NMO operator that provides the GVK of:
```
group: nodemaintenance.medik8s.io
kind: NodeMaintenance
version: v1beta1
```

Signed-off-by: orenc1 <ocohen@redhat.com>

**Reviewer Checklist**

> Reviewers are supposed to review the PR for every aspect below one by one. To check an item means the PR is either "OK" or "Not Applicable" in terms of that item. All items are supposed to be checked before merging a PR. 

- [ ] PR Message
- [ ] Commit Messages
- [ ] How to test
- [ ] Unit Tests
- [ ] Functional Tests
- [ ] User Documentation
- [ ] Developer Documentation
- [ ] Upgrade Scenario
- [ ] Uninstallation Scenario
- [ ] Backward Compatibility
- [ ] Troubleshooting Friendly
  
**Release note**:

```release-note
remove NMO and set it as a dependency to HCO
```

### CODE REVIEW

**Comment by orenc1** [comment]
**NOTE:**

OKD lanes can't be passed until NMO will be released also to the community-operators index, since redhat-operators catalog source is not available by default on OKD clusters.

@razo7 / @slintes , could please provide an estimation for releasing NMO in upstream/community?
The feature freeze is next week.
Thanks

**Comment by razo7** [comment]
> **NOTE:**
> 
> OKD lanes can't be passed until NMO will be released also to the community-operators index, since redhat-operators catalog source is not available by default on OKD clusters.
> 
> @razo7 / @slintes , could please provide an estimation for releasing NMO in upstream/community? The feature freeze is next week. Thanks

I will make a PR to the community-operators today...

**Comment by razo7** [comment]
> > > **NOTE:**
> > > OKD lanes can't be passed until NMO will be released also to the community-operators index, since redhat-operators catalog source is not available by default on OKD clusters.
> > > @razo7 / @slintes , could please provide an estimation for releasing NMO in upstream/community? The feature freeze is next week. Thanks
> > 
> > 
> > I will make a PR to the community-operators today...
> 
> Thank you very much +1

Merged :) https://github.com/redhat-openshift-ecosystem/community-operators-prod/pull/1156

**Comment by orenc1** [comment]
OKD lanes are currently testing OKD 4.9, which its built-in community-operators catalogsource doesn't include NMO, as it is included in 4.10+.
since OKD 4.11 is not available yet, we're bumping OKD in HCO CI to 4.10:
https://github.com/openshift/release/pull/28251

File: controllers/hyperconverged/hyperconverged_controller.go [hunk 1]
```diff
@@ -476,6 +480,33 @@ func (r *ReconcileHyperConverged) handleUpgrade(req *common.HcoRequest, init boo
...
+			ObjectMeta: metav1.ObjectMeta{
+				Name: oldNmoCr.Name,
+			},
+			Spec: nmoapiv1beta1.NodeMaintenanceSpec{
+				NodeName: oldNmoCr.Spec.NodeName,
+				Reason:   oldNmoCr.Spec.Reason,
+			},
+		}
+		err := r.client.Create(req.Ctx, newNmoCr)
+		if err != nil && !strings.Contains(err.Error(), "already exists") {
```
Comment (tiraboschi) [inline controllers/hyperconverged/hyperconverged_controller.go:501]: can you please use IsAlreadyExists(err) from `k8s.io/apimachinery/pkg/api/errors`

Comment (orenc1) [reply controllers/hyperconverged/hyperconverged_controller.go]: fixed, thanks.

**Review by tiraboschi — APPROVED** [review]

**Pull request merged**